src/org/spdx/licensexml/LicenseXmlHelper.java

```diff
@@ -324,6 +339,28 @@ private static void appendOptionalText(Element element,
 		}
 	}
 
+	/**
+	 * @param element parent element
+	 * @return true if the element includes any phrasing content per https://www.w3.org/TR/2014/REC-html5-20141028/dom.html#phrasing-content-1
+	 */
+	private static boolean includesPhrasing(Element element) {
+		NodeList children = element.getChildNodes();
```

Comment:
[`getChildNodes`][1] seems to only return immediate children.  We want something that iterates over all descendants to catch `<optional>…<optional><p>…</p></optional></optional>`.  A quick grep through the docs didn't turn up anything that sounded right, so we may need to implement this locally (or call `includesFlowControl` recursively?).

[1]: https://docs.oracle.com/javase/7/docs/api/org/w3c/dom/Node.html#getChildNodes%28%29

Reply:
It already does a recursive search (line 355)

Reply:
I could move the recursive call outside the if element - not sure if it matters, but it will make it a bit more readable